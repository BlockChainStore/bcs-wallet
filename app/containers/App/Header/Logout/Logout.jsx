// @flow
import React from 'react'
import Power from 'react-icons/lib/md/power-settings-new'

import Tooltip from '../../../../components/Tooltip'
import styles from './Logout.scss'

type Props = {
  logout: Function
}

const Logout = ({ logout }: Props) => (
  <div id='logout' className={styles.logout} onClick={logout}>
    <Tooltip title='Logout'>
      <Power />
    </Tooltip>
  </div>
)

export default Logout
