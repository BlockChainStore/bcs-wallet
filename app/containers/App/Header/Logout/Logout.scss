@import '../../../../styles/variables';

.logout {
    font-size: 1.5em;
    cursor: pointer;
    margin-top: -5px;
    color: $default-text;
    a {
        color: inherit;
    }
}