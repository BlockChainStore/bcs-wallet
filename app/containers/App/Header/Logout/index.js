// @flow
import Logout from './Logout'
import withActions from '../../../../hocs/api/withActions'
import { logoutActions } from '../../../../actions/authActions'

type Props = {
  logout: Function
}

const mapActionsToProps = (actions): Props => ({
  logout: () => actions.request()
})

export default withActions(logoutActions, mapActionsToProps)(Logout)
