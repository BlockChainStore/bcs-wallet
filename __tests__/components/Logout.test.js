import React from 'react'
import { shallow } from 'enzyme'

import Logout from '../../app/containers/App/Header/Logout/Logout'

describe('Logout', () => {
  const logout = jest.fn()

  test('should render without crashing', () => {
    const wrapper = shallow(<Logout logout={logout} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should dispatch logout action when clicked', () => {
    const wrapper = shallow(<Logout logout={logout} />)
    expect(logout.mock.calls.length).toEqual(0)
    wrapper.find('#logout').simulate('click')
    expect(logout.mock.calls.length).toEqual(1)
  })
})
