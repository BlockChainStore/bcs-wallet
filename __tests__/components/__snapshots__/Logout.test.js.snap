// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout should render without crashing 1`] = `
<div
  className="logout"
  id="logout"
  onClick={[Function]}
>
  <Tooltip
    title="Logout"
  >
    <MdPowerSettingsNew />
  </Tooltip>
</div>
`;
